### ISSUE TITLE
/api/jobs/<jobid>/retry fails with 500

### ISSUE THREAD

**Issue by ryansanford** [flywheel-io/core#1248]
Found while testing the tip of master (commit 45bdbd1cddc369badca876b60331c38ee10dfd3a)

@kofalt I wonder if this is related to change #1247 ?

```
2018-06-08 16:40:15      scitran.api                  base.py   414:DEBU from ryansanford@flywheel.io POST /api/jobs/5b1ab060e417a7001c1dad72/retry {} request_id=9a78a6f4-1528476015
2018-06-08 16:40:15      scitran.api                  base.py   393:ERRO Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/webapp2.py", line 570, in dispatch
    return method(*args, **kwargs)
  File "/var/scitran/code/api/api/jobs/handlers.py", line 561, in retry
    for x in j.inputs:
TypeError: 'NoneType' object is not iterable
 request_id=9a78a6f4-1528476015
```

**Comment by kofalt** [comment]
Looks like a job without inputs cannot be retried. Easy fix

**Comment by ehlertjd** [comment]
Closed via #1250
